src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameSession.cs

```diff
@@ -517,13 +530,20 @@ private void UpdateConflictResolutionTask()
 
             var asyncToken = _asyncListener.BeginAsyncOperation(nameof(UpdateConflictResolutionTask));
 
-            _conflictResolutionTask = _allRenameLocationsTask.SafeContinueWithFromAsync(
-               t => t.Result.GetReplacementsAsync(replacementText, optionSet, cancellationToken),
-               cancellationToken,
-               TaskContinuationOptions.OnlyOnRanToCompletion,
-               TaskScheduler.Default);
+            _conflictResolutionTask = ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
+            {
+                // Join prior work before proceeding, since it performs a required state update.
+                // https://github.com/dotnet/roslyn/pull/34254#discussion_r267024593
+                //
+                // The cancellation token is passed to the prior work when it starts, not when it's joined. This is
+                // the equivalent of TaskContinuationOptions.LazyCancellation.
```

Comment:
Ditto as earlier: why maintain the LazyCancellation behavior? It doesn't seem necessary, it's not clear why.

Reply:
I was able to prove that this one does not need lazy cancellation. Updated and commented.